docker/ray-deps/Dockerfile

```diff
@@ -12,6 +12,7 @@ COPY $FIND_LINKS_PATH $FIND_LINKS_PATH
 RUN $HOME/anaconda3/bin/pip --no-cache-dir install --find-links $FIND_LINKS_PATH \
     $(basename $WHEEL_PATH)[all] \
     $(if [ "$AUTOSCALER" = "autoscaler" ]; then echo \
+        "redis >= 3.5.0, < 4.0.0" \
```

Comment:
The versions match the current [requirements.txt](https://github.com/ray-project/ray/blob/84279286df856040c772543e9d961502caaa0a70/python/requirements.txt#L88), where redis is specified for testing purposes.